### PR TITLE
#747 Worker Log desensitization(日志脱敏)

### PR DESCRIPTION
**Pull request by Yeleights**
## What is the purpose of the pull request
*#747 Worker Data Source Log Desensitization*

## Brief change log
  - *Add DATASOURCE_PASSWORD_REGEX to Constants*
  - *Add SensitiveLogUtil, SensitiveDataConverter, SensitiveLogUtilTest, SensitiveDataConverterTest*
  - *Add conversionRule to worker_logback.xml*
## Verify this pull request
This change added tests and can be verified as follows:
  - *Added SensitiveLogUtilTest and SensitiveDataConverterTest to verify the change.*

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/log/SensitiveDataConverter.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    /**
+     * password regex
+     *
+     * @param logMsg original log
+     */
+    private String passwordHandler(String logMsg) {
+
+        Pattern pattern = Pattern.compile(Constants.DATASOURCE_PASSWORD_REGEX);
```
Comment (Technoboy-) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/log/SensitiveDataConverter.java]: put this line as private object

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/log/SensitiveDataConverter.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        Matcher matcher = pattern.matcher(logMsg);
+
+        StringBuffer sb = new StringBuffer();
```
Comment (Technoboy-) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/log/SensitiveDataConverter.java]: StringBuffer sb = new StringBuffer(logMsg.length);

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/log/SensitiveDataConverter.java (hunk 1 above)
Comment (Yeleights) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/log/SensitiveDataConverter.java]: thx, i will fix it.

**Comment by lenboo** [comment]
@Yeleights please resolve the conflicts.

**Pull request merged**